Mathlib/GroupTheory/SpecificGroups/BraidGroup/Basic.lean

```diff
@@ -0,0 +1,83 @@
+/-
+Copyright (c) 2024 Hannah Fechtner. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Hannah Fechtner
+-/
+import Mathlib.GroupTheory.PresentedGroup
+import Mathlib.Data.Nat.Dist
+
+/-!
+#  Braid Groups
+
+Artin's braid group on infintely many strands
+
+## Main definitions
+
+* `BraidGroupInf ` is the braid group on infinitely many strands, defined as a PresentedGroup
+  on the braid relations
+
+## Main results
+* will add in as they get PR'd
+
+## Tags
+braid group
+
+## TODO
+* `braid_group n` is the braid group on n+1 strands (thus n generators)
+
+-/
+
+
+/-- the set of relations on FreeGroup ℕ corresponding to valid braid moves. generators are indexed
+by natural numbers. commutativity holds if the generators are at least 2 apart. if two generators
+a and b are adjacent, then aba is congruent to bab -/
+def braid_rels_inf : Set (FreeGroup ℕ) :=
+  { r | ∃ i j : ℕ , 1 = Nat.dist i j ∧ r = .of i * .of j * .of i *
+    (.of j)⁻¹ * (.of i)⁻¹ * (.of j)⁻¹} ∪
+  { r | ∃ i j : ℕ, 2 ≤ Nat.dist i j ∧ r = .of i * .of j * (.of i)⁻¹ * (.of j)⁻¹}
+
+/-- Artin's braid group on infinitely many strands -/
+def BraidGroupInf := PresentedGroup braid_rels_inf
+
+instance : Group BraidGroupInf := by
+  unfold BraidGroupInf; infer_instance
+
+namespace BraidGroupInf
+
+/-- the ith generator of the infinite braid group, whose geometric interpretation is crossing
+the ith strand over the (i+1)th strand -/
+def σ (k : ℕ) : BraidGroupInf := PresentedGroup.of k
+
+theorem BraidGroupInf.braid {i j : ℕ} (h : 1 = Nat.dist i j): σ i * σ j * σ i =
+    σ j * σ i * σ j := by
+  symm; rw [← mul_inv_eq_one]
+  exact QuotientGroup.eq.mpr <| Subgroup.subset_normalClosure <| Or.inl <| Exists.intro i <|
+    Exists.intro j <| ⟨h, rfl⟩
+
+theorem BraidGroupInf.comm {i j : ℕ} (h : 2 ≤ Nat.dist i j) : σ i * σ j = σ j * σ i := by
```

Comment:
Because of the namespace, you now get duplication in the names
```suggestion
private theorem comm {i j : ℕ} (h : 2 ≤ Nat.dist i j) : σ i * σ j = σ j * σ i := by
```

Reply:
fixed it! I plan to use this theorem outside this file (often), so if it's alright, I'd rather not make it private